src/main/thrift/parquet.thrift

```diff
@@ -902,46 +929,29 @@ struct FileMetaData {
    * The obsolete min and max fields are always sorted by signed comparison
    * regardless of column_orders.
    */
-  7: optional list<ColumnOrder> column_orders;
-}
-
-struct AesGcmV1 {
-  /** Retrieval metadata of AAD used for encryption of pages and structures **/
-  1: optional binary aad_metadata
-
-  /** If file IVs are comprised of a fixed part, and variable parts
-   *  (e.g. counter), keep the fixed part here **/
-  2: optional binary iv_prefix
- 
-}
-
-struct AesGcmCtrV1 {
-  /** Retrieval metadata of AAD used for encryption of structures **/
-  1: optional binary aad_metadata
-
-  /** If file IVs are comprised of a fixed part, and variable parts
-   *  (e.g. counter), keep the fixed part here **/
-  2: optional binary gcm_iv_prefix
-
-  3: optional binary ctr_iv_prefix
-}
-
-union EncryptionAlgorithm {
-  1: AesGcmV1 AES_GCM_V1
-  2: AesGcmCtrV1 AES_GCM_CTR_V1
+  7: optional list<ColumnOrder> column_orders
+  
+  /** 
+   * Set in encrypted files with plaintext footer.
+   * Files with encrypted footer store algorithm id
+   * in FileCryptoMetaData structure.
+   */
+  8: optional EncryptionAlgorithm encryption_algorithm
 }
 
+/** Crypto metadata for files with encrypted footer **/
 struct FileCryptoMetaData {
+  /** 
+   * Files with plaintext footer store algorithm id
```

Comment:
It is a bit misleading that you only document the *other* behavior without mentioning the one when this field is used.

```suggestion
   * Encryption algorithm. Please note that this field is only used for files
   * with encrypted footer. Files with plaintext footer store the algorithm id
```